samples/net/zperf/src/zperf_tcp_uploader.c

```diff
@@ -85,7 +85,7 @@ void zperf_tcp_upload(const struct shell *shell,
 		}
 
 #if defined(CONFIG_ARCH_POSIX)
-		k_busy_wait(K_MSEC(100));
+		k_busy_wait(K_MSEC(100) * USEC_PER_MSEC);
```

Comment:
@jukkar Maybe a silly question, but how come are you doing a k_busy_wait instead of a k_sleep?

Reply:
> Maybe a silly question, but how come are you doing a k_busy_wait instead of a k_sleep?

I think it was mentioned in the native_posix documentation to call `k_busy_wait()`, is this changed?

Reply:
Short answer: Only if the time is too small to k_sleep( ), or, there is some other reason why you don't want to yield. Meaning, you would normally use k_busy_wait() only if you really want to have a busy wait loop.
You can use it here, there is nothing intrinsically wrong, it is just a bit funny looking as in other architectures you are actually yielding, and in the POSIX arch you are busy waiting for 100ms.
That was the reason for my question.

Long answer:
What the documentation tried to clarify: You cannot have infinite loops in the POSIX ARCH without breaking them in some way. Otherwise simulated time does not pass in that infinite loop, and it becomes a real infinite loop where nothing at all happens.
So you can do anything which will cause your thread to let time pass. One option is k_sleep() where you also yield.
Another option, if you want to "emulate" a busy wait loop, is to use a k_busy_wait() with a very small delay (say a couple of µs). In this last case, if the delay is very small, you are adding quite a bit of overhead in the simulation.

Reply:
There are 2 separate issues here: using `k_busy_wait()` and passing the arguments in microseconds. I would like to merge this to get correct code in the tree, and this can be converted into a `k_sleep()` in a different PR.

Reply:
> as in other architectures you are actually yielding

I faintly remember that there was weird issues when just calling `k_yield()` for native_posix here. But it was several months ago so I might remember wrong.

Reply:
@jukkar If you only k_yield() you would still have a busy loop (just going all the way to the kernel and back). Maybe it was that.

@carlescufi This was a sidediscussion, not meant to hold this PR in any way.